AK/StringUtils.h

```diff
@@ -38,6 +39,13 @@ enum class TrimWhitespace {
     No,
 };
 
+enum class OnSplit : unsigned {
+    Nothing = 0,
+    KeepSeparator = 1,
+    KeepEmpty = 2,
+};
+AK_ENUM_BITWISE_OPERATORS(OnSplit);
```

Comment:
There's no way this should an enum with bitwise operators, these are mutually exclusive.

Reply:
That's to be able to do `split_view(..., OnSplit::KeepSeparator | OnSplit::KeepEmpty)`. How should it be done otherwise?

Reply:
That also belongs in a separate commit then, the book -> enum commit should be functionally equivalent and not support multiple flags at first